src/Features/Core/Portable/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs

```diff
@@ -34,6 +34,9 @@ internal abstract class AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer
                                      customTags: WellKnownDiagnosticTags.NotConfigurable);
 
         protected abstract LocalizableString GetTitleAndMessageFormatForClassificationIdDescriptor();
+        protected abstract ImmutableArray<SyntaxNode> MergeImports(ImmutableArray<SyntaxNode> unnecessaryImports);
```

Comment:
`MergeImports` definition is just moved from bottom of the file to at the top with other abstract methods.

`IsRegularCommentOrDocComment` is the new abstract method added here, which can be removed once `ISyntaxFactsService` is available in the shared layer.

Reply:
can you file issue for the latter bit?

Reply:
(note: i'm happy to fix this up, but we should track it).

Reply:
Added a bullet item in https://github.com/dotnet/roslyn/issues/41462